feat(docker): add data persistence to docker-compose (#834)
getfloresta/Floresta: doc/docker.md

Floresta also (optionally) provides [Prometheus](https://prometheus.io/) metrics endpoint, which you can enable at compile time. If you want a quick setup with Grafana, we provide a [docker-compose.yml](../docker-compose.yml) for that as well. Just use:

```bash
docker compose up -d
```

This will start Floresta on Bitcoin mainnet by default. All blockchain data, metrics, and Grafana configurations are persisted in Docker volumes.

## Running on Different Networks

The provided `docker-compose.yml` supports running Floresta on different Bitcoin networks. To make this easy and avoid port collisions, we provide a sample environment file.

Copy the sample file from `contrib/env.docker.sample`:

```bash
cp contrib/env.docker.sample .env
```

Edit the `.env` file to uncomment the network you want to use (which sets the correct NETWORK, RPC_PORT, and ELECTRUM_PORT), and then run:

```bash
docker compose up -d
```

Alternatively, you can pass the variables directly inline:

```bash
NETWORK=signet RPC_PORT=38332 ELECTRUM_PORT=60001 docker compose up -d
```

## Using Local Floresta Data

> If you already have Floresta data on your machine (e.g., from running it natively), you can reuse that data instead of starting from scratch:

Reviewer: ```suggestion
If you already have a florestad datadir on your machine (e.g., from a previous run), you can reuse that datadir instead of starting from scratch:
```

```bash
# Use your existing ~/.floresta directory
FLORESTA_DATA=$HOME/.floresta docker compose up -d
```

You can also uncomment and set the FLORESTA_DATA variable directly inside your `.env` file:

```yml

> FLORESTA_DATA=/home/your-user/.floresta

Reviewer: ```suggestion
FLORESTA_DATA=$HOME/.floresta
```
